bin/BuildPackages.sh

```diff
@@ -79,115 +135,197 @@ build_carat() {
 # It is not possible to move around compiled binaries because these have the
 # path to some data files burned in.
 zcat carat-2.1b1.tgz | tar pxf -
-ln -s carat-2.1b1/bin bin
+# If the symbolic link does not exist then create it.
+# Does not check if bin exists already with some content or as a file, etc.
+# Thus this code is unstable, but should be corrected by package author
+# and not by this script.
+if [[ ! -L ./bin ]]
+then
+  ln -s ./carat-2.1b1/bin ./bin
+fi
 cd carat-2.1b1
-make TOPDIR=`pwd`
+make TOPDIR="$(pwd)"
 chmod -R a+rX .
 cd bin
-aa=`./config.guess`
-for x in "`ls -d1 $GAPDIR/bin/${aa}*`"; do
- ln -s "$aa" "`basename $x`"
-done
+# This sets GAParch as it should be.
+# Alternatively, one could check $GAPDIR/bin for all directories instead.
+source "$GAPDIR/sysinfo.gap"
+# If the symbolic link does not exist then create it.
+if [[ ! -L ./"$GAParch" ]]
+then
+  shopt -s nullglob
+  # We assume that there is only one appropriate directory....
+  for archdir in *
+  do
+    if [[ -d "$archdir" ]] && [[ ! -L "$archdir" ]]
+    then
+      ln -s ./"$archdir" ./"$GAParch"
+    fi
+  done
+  shopt -u nullglob
+fi
```

Comment:
I will be honest with you, the README of the package is not very clear. I do not really understand why this is not done by ````carat````'s own ````make````. But anyway, in most cases there will be only one directory here, and then a symlink (named ````$GAParch````) will be created to this one directory. I did not use ````ls```` or ````find````, because their output is not entirely reliable (although probably would suffice here).

If there are more directories here, then yes, there will be an error. However, I can only imagine more directories here if somebody is an expert and compiled ````carat```` using several different compilers. But then they should know what they are doing and should not care about ````BuildPackages.sh```` doing the building for this particular package, but rather do the symlinking themselves.

In any case, it would be nice if somebody from ````carat````'s developers/maintainers could comment on this so we could have a more foolproof solution.